purchase_work_acceptance_late_fine/models/account_move.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+    _description = "Journal Entries"
+
+    late_wa_id = fields.Many2one(
+        comodel_name="work.acceptance",
+        string="Late WA",
+        domain=lambda self: self._get_fine_amount(),
+    )
+
+    def _get_fine_amount(self):
+        wa = self.env["work.acceptance"].search([("late_fine", ">", 0)])
+        return [("id", "in", wa.ids)]
+
+    @api.onchange("late_wa_id")
+    def _onchange_wa_id(self):
+        self.line_ids = False
+        # Copy partner.
+        self.partner_id = self.late_wa_id.partner_id
+        wa_lines = self.late_wa_id
+        new_lines = self.env["account.move.line"]
+        if self.type == "out_invoice":
+            for line in wa_lines:
+                amount_penalty = self.late_wa_id.late_fine
+                new_line = new_lines.new(line._prepare_account_move_line(self))
+                new_line.account_id = self.env.company.wa_late_fine_account_id
+                new_line.price_unit = amount_penalty
+                new_line.name = "รายได้ค่าปรับส่งของล่าช้า WA เลขที่ %s" % (line.name)
```

Comment:
```suggestion
                new_line = new_lines.new(line._prepare_account_move_line(self))
```
Everything should be in function _prepare_account_move_line(). This is the point of having sub function.